core/src/main/java/org/apache/calcite/rel/metadata/RelMdDistribution.java

```diff
@@ -141,10 +141,17 @@ public static RelDistribution limit(RelMetadataQuery mq, RelNode input) {
    * {@link org.apache.calcite.rel.core.Calc}'s distribution. */
   public static RelDistribution calc(RelMetadataQuery mq, RelNode input,
       RexProgram program) {
-    throw new AssertionError(); // TODO:
+    assert program.getCondition() != null || !program.getProjectList().isEmpty();
```

Comment:
What's the assertion for ?
It seems common case that program.getCondition is null when there's no filtering condition.

Reply:
Make sure that the program is valid.
As my understanding, empty project list and null condition cannot come together.